### PR TITLE
Shib & Cas Authentification method

### PR DESCRIPTION
**Pull request by Barbapapou**
Authentication par Cas et Shibboleth dans le cadre de l'intégration de CorrectExam au sein de la DSI.

### CODE REVIEW

File: src/main/docker/Dockerfile.build.amd64
Comment (barais) [inline src/main/docker/Dockerfile.build.amd64:1]: These changes must not be in the pull request

File: src/main/docker/Dockerfile.front
Comment (barais) [inline src/main/docker/Dockerfile.front:1]: These changes must not be in the pull request

File: src/main/docker/app.yml
Comment (barais) [inline src/main/docker/app.yml:1]: These changes must not be in the pull request

Comment (barais) [inline src/main/docker/app.yml:1]: These changes must not be in the pull request

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
-%devold.quarkus.datasource.password=
-%dev.quarkus.datasource.jdbc.url=jdbc:mysql://localhost:3306/gradeScopeIstic?useUnicode=true&characterEncoding=utf8&useSSL=false
+%dev.quarkus.datasource.jdbc.url=jdbc:mysql://localhost:3306/correctexam?useUnicode=true&characterEncoding=utf8&useSSL=false&allowPublicKeyRetrieval=true
 %dev.quarkus.datasource.db-kind=mysql
-%dev.quarkus.datasource.username=gradescope
-%dev.quarkus.datasource.password=test
+%dev.quarkus.datasource.username=root
+%dev.quarkus.datasource.password=
 %dev.quarkus.hibernate-orm.log.sql=false
+configcas.server_cas = https://sso-cas6-test.univ-rennes1.fr
```
Comment (barais) [inline src/main/resources/application.properties:9]: Does it mean that if this key are absent, we go back to a standard authentification ?

File: src/main/docker/Dockerfile.front
Comment (barais) [inline src/main/docker/Dockerfile.front:1]: Keep the default url in dockerfile

File: src/main/docker/app.yml
Comment (barais) [inline src/main/docker/app.yml:1]: do not touch the app.yml in your PR

File: src/main/docker/exampleconf/exam.conf
Comment (barais) [inline src/main/docker/exampleconf/exam.conf:1]: Do not force a specific https on localhost

File: src/main/resources/config/liquibase/changelog/20220719165535_update_entity_StudentResponse.xml
Comment (barais) [inline src/main/resources/config/liquibase/changelog/20220719165535_update_entity_StudentResponse.xml:1]: why did you remove this

File: src/main/resources/config/liquibase/changelog/20220824165535_update_entity_Question.xml
Comment (barais) [inline src/main/resources/config/liquibase/changelog/20220824165535_update_entity_Question.xml:1]: why did you remove this

**Review by barais — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties (hunk 1 above)
Comment (Barbapapou) [reply src/main/resources/application.properties:9]: CAS authentication is simply reached from another endpoint. The usual way of authentication is still available regardless of the state of this key.

File: src/main/resources/config/liquibase/changelog/20220719165535_update_entity_StudentResponse.xml
Comment (Barbapapou) [reply src/main/resources/config/liquibase/changelog/20220719165535_update_entity_StudentResponse.xml:1]: While attempting to set up a MySQL database, I encountered error messages indicating that this option is not available. This doesn't change the effect of the migration.

**Comment by Barbapapou** [comment]
The requested changes have been made.

**Pull request merged**